packages/dev/tsconfig/README.md

```diff
@@ -0,0 +1,139 @@
+# @lokalise/tsconfig
+
+Shared TypeScript configuration for Lokalise projects.
+
+## Getting Started
+
+### Requirements:
+- TypeScript `^5.8.0`
+- ESM codebase
+
+### Installation:
+```bash
+npm install --save-dev @lokalise/tsconfig
+```
+
+TypeScript configuration depends on whether you're using `tsc` to build your code or a `bundler` (Vite, esbuild, SWC, etc.).
+
+## **Using a Bundler**
+
+Create a `tsconfig.json` for type-checking:
+
+```jsonc
+{
+  "extends": "@lokalise/tsconfig/bundler",
+  "extends": "@lokalise/tsconfig/bundler-dom", // When your code runs in the DOM
+  // Whitelist files that should be used for type-checking:
+  "include": ["src/**/*", "test/**/*", "vitest.config.ts"]
+}
+```
+**Note:** Choose only one `"extends"` option for `tsconfig.json`.
+
+## **Using `tsc` (TypeScript Compiler)**
```

Comment:
we can make it more explicit by mentioning that this is typically used for backend applications and libraries